src/EFCore.PG/Design/Internal/NpgsqlAnnotationCodeGenerator.cs

```diff
@@ -88,6 +86,20 @@ public override MethodCallCodeFragment GenerateFluentApi(IModel model, IAnnotati
                         enumTypeDef.Schema, enumTypeDef.Name, enumTypeDef.Labels);
             }
 
+            if (annotation.Name.StartsWith(NpgsqlAnnotationNames.RangePrefix))
+            {
+                var rangeTypeDef = new PostgresRange(model, annotation.Name);
+
+                return new MethodCallCodeFragment(nameof(NpgsqlModelBuilderExtensions.ForNpgsqlHasRange),
```

Comment:
Thanks for doing this. One thought is to check whether the more advanced/exotic fields are null, and produce a simplified `ForNpgsqlHasRange()` call. I'd wager 99% of range definitions will have the last 4 fields empty...

Reply:
Good idea. I've updated it to consider when the optionals are all null.